packages/devtools_app/lib/src/screens/inspector_v2/widget_properties/properties_view.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+import 'package:logging/logging.dart';
+
+import '../../../shared/diagnostics/diagnostics_node.dart';
+import '../../../shared/diagnostics_text_styles.dart';
+import '../inspector_controller.dart';
+
+final _log = Logger('properties_view');
+
+class PropertiesView extends StatelessWidget {
+  const PropertiesView({
+    super.key,
+    required this.controller,
+    required this.node,
+  });
+
+  final InspectorController controller;
+  final RemoteDiagnosticsNode node;
+
+  Future<List<RemoteDiagnosticsNode>> loadProperties() async {
+    final properties = <RemoteDiagnosticsNode>[];
+    final objectGroupApi = node.objectGroupApi;
+    if (objectGroupApi == null) return Future.value(properties);
+    try {
+      // Fetch widget properties:
+      final widgetProperties = await node.getProperties(objectGroupApi);
+      properties.addAll(widgetProperties);
+      // Fetch RenderObject properties:
+      for (final widgetProperty in widgetProperties) {
+        if (widgetProperty.propertyType == 'RenderObject') {
+          final renderProperties =
+              await widgetProperty.getProperties(objectGroupApi);
+          // Only display RenderObject properties that are not already set on
+          // the widget:
+          for (final renderProperty in renderProperties) {
+            final propertyOnWidget = widgetProperties
+                    .firstWhereOrNull((p) => p.name == renderProperty.name) !=
+                null;
+            if (!propertyOnWidget) {
+              properties.add(renderProperty);
+            }
+          }
+        }
+      }
+      return Future.value(properties);
+    } catch (e, st) {
+      _log.warning(e, st);
+      return Future.value(properties);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<List<RemoteDiagnosticsNode>>(
+      // ignore: discarded_futures, FutureBuilder requires a future.
+      future: loadProperties(),
+      builder: (context, snapshot) {
+        final properties = snapshot.data ?? <RemoteDiagnosticsNode>[];
+        return Container(
+          margin: const EdgeInsets.all(denseSpacing),
+          child: Scrollbar(
+            child: SingleChildScrollView(
+              primary: true,
+              child: Table(
```

Comment:
This will not be lazy. Can we use a ListView.builder here instead of a SingleChildScrollView with a Table widget?

Reply:
Done!